gazelle/python/generate.go

```diff
@@ -239,16 +236,34 @@ func (py *Python) GenerateRules(args language.GenerateArgs) language.GenerateRes
 			}
 		}
 
-		pyLibrary = newTargetBuilder(pyLibraryKind, pyLibraryTargetName, pythonProjectRoot, args.Rel, pyFileNames).
+		pyLibrary := newTargetBuilder(pyLibraryKind, pyLibraryTargetName, pythonProjectRoot, args.Rel, pyFileNames).
 			addVisibility(visibility).
-			addSrcs(pyLibraryFilenames).
+			addSrcs(srcs).
 			addModuleDependencies(deps).
 			generateImportsAttribute().
 			build()
 
 		result.Gen = append(result.Gen, pyLibrary)
 		result.Imports = append(result.Imports, pyLibrary.PrivateAttr(config.GazelleImportsKey))
 	}
+	if cfg.PerFileGeneration() {
+		pyLibraryFilenames.Each(func(index int, filename interface{}) {
+			if filename == pyLibraryEntrypointFilename {
+				stat, err := os.Stat(filename.(string))
+				if err != nil {
+					log.Fatalf("ERROR: %v\n", err)
+				}
+				if stat.Size() == 0 {
+					return // ignore empty __init__.py
+				}
+			}
+			srcs := treeset.NewWith(godsutils.StringComparator, filename)
+			pyLibraryTargetName := strings.TrimSuffix(filepath.Base(filename.(string)), ".py")
```

Comment:
Would it be possible to detect if the file has a `if __name__ == "__main__":` and then generate a `py_binary` target for such a file?

Reply:
it's... possible, but I dunno how desirable that would be. a lot of times, it's either just demo code for a library that rarely needs to be executed or it's test harness bootstrap. I think we'd be better off detecting if it was marked executable (u+x), though I dunno how that'd work on stuff like NTFS

in any case, can we leave that to a future PR?